samples/web/content/apprtc/probers.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/python2.4
+#
+# Copyright 2014 Google Inc. All Rights Reserved.
+
+"""AppRTC Probers
+
+This module implements CEOD and collider probers.
+"""
+
+import constants
+import logging
+import json
+import webapp2
+from google.appengine.api import mail
+from google.appengine.api import urlfetch
+
+def sendAlertEmail(receiver, tag, message):
+  if not mail.is_email_valid(receiver):
+    log.warning('Invalid receiver email: ' + receiver)
+    return
+
+  sender_address = 'AppRTC Notification <jiayl@google.com>'
+  subject = 'AppRTC Prober Alert: ' + tag
+  body =  """
+  AppRTC Prober detected an error:
+
+  %s
+  """ % message
+
+  logging.info('Sending email to %s: subject=%s, message=%s' \
+      % (receiver, subject, message))
+  mail.send_mail(sender_address, receiver, subject, body)
+
+def has_non_empty_string_value(dict, key):
+  return key in dict and \
+         isinstance(dict[key], basestring) and \
+         dict[key] != ''
+
+def has_non_empty_array_value(dict, key):
+  return key in dict and \
+         isinstance(dict[key], list) and \
+         len(dict[key]) > 0
+
+class ProbeCEODPage(webapp2.RequestHandler):
+  def get(self):
+    alert_email = self.request.get('email')
```

Comment:
that is, to an arbitrary user. I think we probably want to have the target email address stored in the code or in a DB somewhere, not passed in from the URL.